docker/studyu-cli.sh

```diff
@@ -0,0 +1,479 @@
+#!/bin/bash
```

Comment:
macOS sadly has a really old version of bash at `/bin/bash`. I would suggest using `/usr/bin/env bash` for the shebang which will use a newer version (if the user has one) and including a version check before running because otherwise people may get error messages they don't understand, such as

```plain
./studyu-cli.sh: line 17: declare: -A: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
```

for `GNU bash, version 3.2.57(1)-release (arm64-apple-darwin22)`

Reply:
👍 Do you think something like this would be sufficient?

```
version_above_4(){
    [ -z $BASH_VERSION ] && return 1
    case $BASH_VERSION in 
        4.*) return 0 ;;
        ?) return 1;; 
    esac
}
if  ! version_above_4; then
    echo "Bash version equal or larger than 4 is required to run this script."
fi
```

Reply:
Like this it would fail for versions > 4, is that intended? I only have that old bash version and `5.2.15(1)-release` on my machine. Anything starting at version 4 is okay, I would suggest

```bash
if [[ "$(head -c 1 <<< $BASH_VERSION)" -lt 4 ]]; then
    echo "Bash version equal or larger than 4 is required to run this script."
    exit 1
fi
```

This also still covers the case that bash is not the shell running the script since `"" < 4` :)